File: back/src/models/supplier.model.ts
// src/models/supplier.model.ts
import { Model, DataTypes } from 'sequelize';
import sequelize  from '../config/mysql';

class Supplier extends Model {
  public SupplierID!: number;
  public PersonID!: number;
  public Company!: string;
  public ProductType!: string;
  public Rating!: number;
}

Supplier.init(
  {
    SupplierID: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
    },
    PersonID: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'Person',
        key: 'PersonID'
      }
    },
    Company: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    ProductType: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    Rating: {
      type: DataTypes.INTEGER,
      allowNull: true,
    }
  },
  {
    sequelize,
    modelName: 'Supplier',
    tableName: 'Supplier',
    timestamps: false
  }
);

export default Supplier;

File: front/lib/model/supplier_model.dart
class Supplier {
  final int supplierID;
  final int personID;
  final String company;
  final String productType;
  final int rating;

  Supplier({
    required this.supplierID,
    required this.personID,
    required this.company,
    required this.productType,
    required this.rating,
  });

  factory Supplier.fromJson(Map<String, dynamic> json) {
    return Supplier(
      supplierID: json['supplierID'],
      personID: json['personID'],
      company: json['company'],
      productType: json['productType'],
      rating: json['rating'],
    );
  }
}
